client/src/components/Workflow/Editor/Forms/FormTool.vue

```diff
@@ -124,6 +121,12 @@ export default {
                     }
                 }
             });
+            Utils.deepeach(inputs, (input) => {
+                if (input.type === "conditional") {
+                    input.connectable = false;
+                    input.test_param.collapsible_value = undefined;
+                }
+            });
```

Comment:
I confirm this works, but can you explain why this needs to go into a second deepeach ? At first I thought it's important to set `input.collapsible_value` and `input.is_workflow` for conditional inputs as well, but that's not it, otherwise this would work as well (and save the second deepeach):
```diff
diff --git a/client/src/components/Workflow/Editor/Forms/FormTool.vue b/client/src/components/Workflow/Editor/Forms/FormTool.vue
index ee4f46182a..3873d44c5e 100644
--- a/client/src/components/Workflow/Editor/Forms/FormTool.vue
+++ b/client/src/components/Workflow/Editor/Forms/FormTool.vue
@@ -119,12 +119,10 @@ export default {
                             (input.options && input.options.length === 0) ||
                             ["integer", "float"].indexOf(input.type) != -1;
                     }
-                }
-            });
-            Utils.deepeach(inputs, (input) => {
-                if (input.type === "conditional") {
-                    input.connectable = false;
-                    input.test_param.collapsible_value = undefined;
+                    if (input.type === "conditional") {
+                        input.connectable = false;
+                        input.test_param.collapsible_value = undefined;
+                    }
                 }
             });
             return inputs;
```
I've been staring at this for some time, even printing out the params and diffing them, and I don't understand why this is not equivalent. Is this a disguised case of manipulating props (node in this case) ?

Reply:
Deepeach recursively traverses to each of the nested objects in inputs. At first it reaches the conditional `input` which we use to configure the child-object `input.test_param`. This child-object will be visited by the recursion in consecutive iteration. When it visits the input object of the `test_param` it treats it as a regular input parameter. This is why the two loops have been separated. Alternatively, we could set a flag within the `test_param` object to tag it as conditional test parameter.

Reply:
> it treats it as a regular input parameter.

... and resets connectable to true. Alright that makes sense. I don't know about adding another flag, probably premature optimization.

Is it possible that in a future iteration the form parameters themselves maintain the state in a way that it is directly consumable by the workflow form ?

Reply:
We could just set something like `input.test_param._fixed = true` and then recognize this flag when the `test_param` is visited by itself.

Reply:
Performance-wise that is ok and what you suggested before, but I'd call it premature optimization until we know this is an issue. At most it's going to be twice as slow, that's not a big deal. My concern is more with manipulating the config form outside of the parameters and adding state to it.

IOW I have the intuition that the state should be kept with the parameters. We're special casing a few parameter types here, right ? And all this does is inform how the parameters are going to render. So if you make the Parameters accept a prop that defines whether we're in a workflow or regular tool form scenario the parameters themselves can decide if they have to display workflow specific elements. I hope that makes sense.